distrobuilder/chroot.go

```diff
@@ -176,6 +176,8 @@ func getOverlay(cacheDir, sourceDir string) (func(), string, error) {
 	}
 
 	cleanup := func() {
+		unix.Sync()
```

Comment:
What's that for?

Reply:
If distrobuilder failed for some reason, it would also say that the overly was busy and couldn't be unmounted. This sync removes that additional error.

Reply:
I guess we could also use something like TryUnmount like LXD does.

Reply:
Hmm, but we don't use overlay for the ISO build work, do we?

Feels like this should be move to its own commit.

Reply:
We do. We use the read-only iso as the lowerdir, and modify the .wim files before running `genisoimage` without having to copy the entire iso in order to make it rw.

Reply:
Ah, oops, I missed that :)

Still feels like this should be a separate commit given that it's not really specific to the build-iso stuff.